src/components/ProjectSearch/index.js

```diff
@@ -20,84 +23,256 @@ import {
+      const { setExpanded, file, expanded, match } = this.focusedItem;
+      if (setExpanded) {
+        setExpanded(file, !expanded);
+      } else if (match) {
+        this.selectMatchItem(match);
+      }
+    }
+  };
+
+  inputOnChange = (e: SyntheticInputEvent<HTMLInputElement>) => {
+    const inputValue = e.target.value;
+    this.setState({ inputValue });
+  };
+
+  renderFile(
+    file: Result,
+    focused: boolean,
+    expanded: boolean,
+    setExpanded: Function
+  ) {
+    if (focused) {
+      this.focusedItem = { setExpanded, file, expanded };
+    }
+
+    const matchesLength = file.matches.length;
+    const matches = ` (${matchesLength} match${matchesLength > 1 ? "es" : ""})`;
+
+    return (
+      <div
+        className={classnames("file-result", { focused })}
+        key={file.sourceId}
+        onClick={e => setExpanded(file, !expanded)}
+      >
+        <Svg name="arrow" className={classnames({ expanded })} />
+        <img className="file" />
+        <span className="file-path">{getRelativePath(file.filepath)}</span>
+        <span className="matches-summary">{matches}</span>
+      </div>
+    );
   }
 
-  isProjectSearchEnabled() {
-    return this.props.activeSearch === "project";
+  renderMatch(match: Match, focused: boolean) {
+    if (focused) {
+      this.focusedItem = { match };
+    }
+    return (
+      <div
+        className={classnames("result", { focused })}
+        onClick={() => setTimeout(() => this.selectMatchItem(match), 50)}
+      >
+        <span className="line-number" key={match.line}>
+          {match.line}
+        </span>
+        {this.renderMatchValue(match)}
+      </div>
+    );
   }
 
-  renderTextSearch() {
-    const {
-      sources,
-      results,
-      status,
-      searchSources,
-      closeProjectSearch,
-      selectLocation,
-      textSearchQuery
-    } = this.props;
+  renderMatchValue(lineMatch: Match) {
+    return highlightMatches(lineMatch);
+  }
+
+  renderResults() {
+    const results = this.getResults().filter(
+      result => result.matches.length > 0
+    );
+
+    const { status } = this.props;
+
+    function getFilePath(item: Match, index?: number) {
+      return item.filepath
+        ? `${item.sourceId}-${index || "$"}`
+        : `${item.sourceId}-${item.line}-${item.column}-${index || "$"}`;
+    }
+
+    const renderItem = (item, depth, focused, _, expanded, { setExpanded }) => {
+      return item.filepath
+        ? this.renderFile(item, focused, expanded, setExpanded)
+        : this.renderMatch(item, focused);
+    };
+    if (results.length && status === statusType.done) {
+      return (
+        <ManagedTree
+          getRoots={() => results}
+          getChildren={file => file.matches || []}
+          itemHeight={24}
+          autoExpandAll={true}
+          autoExpandDepth={1}
+          getParent={item => null}
+          getPath={getFilePath}
+          renderItem={renderItem}
+        />
+      );
+    } else if (
```

Comment:
minor thing, but it'd be nice to have the fetching check first... 

also, we dont need the else if... could just be `if`